### PR TITLE
[W-13833522] mtf coverage

### PR DESCRIPTION
**Pull request by cristianchan-mulesoft**
_Hi, thank you for your work. We understand that you want to merge your changes and move on from this issue, but we also want to maintain the safety, readability, and testability of our code. Because of this, we kindly ask that you confirm that you have fulfilled the prerequisites for each category of activity before merging your changes._

### If the Pull Request has a dependency update:

- [ ] I have read the Release Notes for the new version (and intermediate versions if the jump would include more than one). _Don't blindly trust that the dependency will honor SEMVER, always read carefully the release notes_
- [ ] Java 8+ support is maintained
- [ ] The new version has no vulnerabilities
- [ ] A team member reviewed the Pull Request.
- [ ] No backward compatibility is broken.
- [ ] If it is a "provided" dependency, it has been checked that if it is suggested, the version number is the same.

### For ALL the Releases:

- [ ] There is a task in GUS for this change.
- [ ] The corresponding Release Notes have been written and shared with the documentation team.
- [ ] The new code has tests and they have before and after methods to be sure that you are working with a clean environment and that you are leaving the environment clean after they finish.
- [ ] If you are using reflection, create a test to call the methods you are invoking, this way, if there is a change in the API, the test will be able to detect it.
- [ ] If the project has a parent pom and modules, the release pipeline only uploads the modules, so, please, make sure that the parent was deployed in the repositories before considering this release finished.
- [ ] Notify in the Slack Channel that the release is complete, so the Docs team can merge the Documentation and Release Notes PR. The docs team always checks in Anypoint Exchange that the connector is released before merging the release notes.
- [ ] Add the connector build to GUS, and assign that build to the GUS ticket.
- [ ] Create a change case request, based on the Change Case Management doc https://docs.google.com/document/d/1tMJlTTZLaXMLiYwxlMBFY1yJwBmejhc4-IP8HAXgDfY/edit#

**NOTE**: _(applies only for Core-Connectors connectors and modules) The release process ends when you merge the pull request that updates the support branch to the new snapshot, please don't forget to do this._

### Patch Version:

- [ ] The Pull Request has been peer-reviewed.
- [ ] No backward compatibility is broken.
- [ ] Documentation: (Required) Release Notes PR with compatibility table. Share it with the Docs team.

### Minor Version:

- [ ] Have a document with the specification of the release.
- [ ] Create a slack channel (eg: rl-conninfra-sftp-1.1.1-new-chiper) to coordinate the release of this version with the documentation team and the architects of our team (at least one week before the release). Share the spec doc with the docs team.
- [ ] The documentation fork has been done and has been updated with the changes related to the new functionalities.
- [ ] The pull request has been reviewed by a peer and the team leader.
- [ ] No backward compatibility is broken.
- [ ] Public Documentation: Minor releases nearly always require a revision of the documentation because the release usually includes new features as well as bug fixes. Even if there are no user-facing changes, documentation has to be versioned for every minor release.
  - [ ] (Required) Release Notes PR with compatibility table. Share it with the Docs team.
  - [ ] (Likely required) Reference guide. Run the script and share the file with the Docs team.
  - [ ] (Assess) Often a minor release impacts UI and examples, so the documentation (user guides, examples, troubleshooting guides) should be reviewed and updated accordingly. Notify the docs team if any updates are needed.

### Major Version:

- [ ] Have a document with the specification of the release.
- [ ] Create a slack channel (eg: rl-conninfra-sftp-2.0.0) to coordinate the release of this version with the documentation team and the architects of our team (at least one month before the release). Share the spec doc with the docs team.
- [ ] The documentation fork has been made and updated with the changes related to the new functionalities and possible compatibility problems with the previous version.
- [ ] The systems properties have been removed, detailing in the documentation the changes in the default behavior of the product.
- [ ] The TODO comments of the code have been reviewed.
- [ ] A colleague, the team leader, and the team architect have reviewed the Pull Request.
- [ ] Public Documentation: Major releases break backward compatibility and should never be released without documentation, as this breaks the user experience.
  - [ ] (Required) Upgrade and Migration Guide: Every major release must have a detailed upgrade guide (template) with information about what was changed and what steps the user has to take to ensure the connector works as expected. Share the details with the Docs team so they can update the guide.
  - [ ] (Required) Release Notes PR with compatibility table. Share it with the docs team.
  - [ ] (Required) Reference guide. Run the script and share the file with the Docs team.
  - [ ] User guide updates, new screenshots or examples (if applicable), share the updates with the docs team.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -990,12 +1015,48 @@
                                 <version>${mtf-tools.version}</version>
                                 <classifier>mule-plugin</classifier>
                             </dependency>
+                            <dependency>
+                                <groupId>org.apache.commons</groupId>
+                                <artifactId>commons-dbcp2</artifactId>
+                                <version>2.9.0</version>
```
Comment (sabecasismulesoft) [inline pom.xml:1021]: Can we please move this version number to a property?

File: pom.xml [hunk 2]
```diff
@@ -990,12 +1015,48 @@
...
                                 <classifier>mule-plugin</classifier>
                             </dependency>
+                            <dependency>
+                                <groupId>org.mule.modules</groupId>
+                                <artifactId>mule-spring-module</artifactId>
+                                <version>${mule-spring-module-version}</version>
+                                <classifier>mule-plugin</classifier>
+                            </dependency>
+                            <dependency>
+                                <groupId>org.springframework</groupId>
```
Comment (sabecasismulesoft) [inline pom.xml:1036]: Are these spring dependencies supposed to be test scope?

File: src/docker/derby/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+FROM eclipse-temurin:17-jre
+
+ENV DERBY_VERSION=10.16.1.1
+ENV DERBY_HOME=/derby
+ENV DERBY_LIB=${DERBY_HOME}/lib
+ENV CLASSPATH=${DERBY_LIB}/derby.jar:${DERBY_LIB}/derbynet.jar:${DERBY_LIB}/derbytools.jar:${DERBY_LIB}/derbyoptionaltools.jar:${DERBY_LIB}/derbyclient.jar
+
+RUN \
+    apt-get update && apt-get install -y wget netcat && \
+    wget https://dist.apache.org/repos/dist/release/db/derby/db-derby-${DERBY_VERSION}/db-derby-${DERBY_VERSION}-bin.tar.gz && \
```
Comment (sabecasismulesoft) [inline src/docker/derby/Dockerfile:10]: This is not a good idea, we may run into the rate limit issue we had with JMS connector, I think we should be uploading this tar.gz to nexus and downloading it with maven to be later used by docker. You can create a follow up ticket for this, as an outcome for this comment. But that change has to be performed eventually.

File: src/test/munit/oracle/connection-oracle-test-case.xml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      xmlns:db="http://www.mulesoft.org/schema/mule/db"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:munit="http://www.mulesoft.org/schema/mule/munit"
+      xmlns:mtf="http://www.mulesoft.org/schema/mule/mtf"
+      xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+        http://www.mulesoft.org/schema/mule/munit http://www.mulesoft.org/schema/mule/munit/current/mule-munit.xsd
+		http://www.mulesoft.org/schema/mule/db http://www.mulesoft.org/schema/mule/db/current/mule-db.xsd
+		http://www.mulesoft.org/schema/mule/mtf http://www.mulesoft.org/schema/mule/mtf/current/mule-mtf.xsd">
+
+    <munit:config name="connection-oracle-test-case.xml" minMuleVersion="4.4.0">
```
Comment (sabecasismulesoft) [inline src/test/munit/oracle/connection-oracle-test-case.xml:13]: Why is this test 4.4.0 onwards?

**Review by sabecasismulesoft — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (cristianchan-mulesoft) [reply pom.xml:1036]: Hello @sabecasismulesoft  yes is just for testing purpose

File: src/docker/derby/Dockerfile (hunk 3 above)
Comment (cristianchan-mulesoft) [reply src/docker/derby/Dockerfile:10]: Ok I will create the ticket

File: pom.xml (hunk 1 above)
Comment (cristianchan-mulesoft) [reply pom.xml:1021]: Done

File: src/docker/derby/Dockerfile (hunk 3 above)
Comment (cristianchan-mulesoft) [reply src/docker/derby/Dockerfile:10]: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001h6g0VYAQ/view

File: src/test/munit/oracle/connection-oracle-test-case.xml (hunk 4 above)
Comment (cristianchan-mulesoft) [reply src/test/munit/oracle/connection-oracle-test-case.xml:13]: Changing this to mule 4.3

**Review by sabecasismulesoft — APPROVED** [review]

**Pull request merged**